x-pack/plugins/monitoring/public/application/hooks/use_local_storage.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState } from 'react';
+import { has } from 'lodash';
+
+interface ParsedMonitoringData {
+  [key: string]: any;
+}
+export const useLocalStorage = <Value>(key: string, defaultValue: Value): [Value, Function] => {
```

Comment:
this hook replaces the "features" service https://github.com/elastic/kibana/blob/master/x-pack/plugins/monitoring/public/services/features.js